osidb/models.py

```diff
@@ -1096,10 +1096,36 @@ class Meta:
     def __str__(self):
         return str(self.uuid)
 
+    def _validate_major_incident_combos(self):
+        if self.type not in (
+            self.FlawMetaType.MAJOR_INCIDENT,
+            self.FlawMetaType.MAJOR_INCIDENT_LITE,
+        ):
+            return
+
+        INVALID_COMBOS = [("+", "+"), ("+", "?"), ("?", "+"), ("?", "-"), ("-", "?")]
+        maj_incident_flag = None
+        maj_incident_lite_flag = None
+
+        for meta in self.flaw.meta.all():
+            if meta.type == FlawMeta.FlawMetaType.MAJOR_INCIDENT:
+                maj_incident_flag = meta.meta_attr.get("status")
+            if meta.type == FlawMeta.FlawMetaType.MAJOR_INCIDENT_LITE:
+                maj_incident_lite_flag = meta.meta_attr.get("status")
+            if maj_incident_flag and maj_incident_lite_flag:
+                break
+
+        flag_pair = (maj_incident_flag, maj_incident_lite_flag)
+        if flag_pair in INVALID_COMBOS:
+            raise ValidationError(
+                f"Flaw MAJOR_INCIDENT and MAJOR_INCIDENT_LITE combination cannot be {flag_pair}."
+            )
+
     def validate(self, *args, **kwargs):
         """validate model"""
         # add custom validation here
         super().clean_fields(*args, exclude=["meta_attr"], **kwargs)
+        self._validate_major_incident_combos()
```

Comment:
Just a thought. If someone would modify the metas on flaw from the flaw side, eg. like flaw.meta.set(...), then this validation is not triggered at all since FlawMeta is basically not saved. I am not sure if we currently do this kind of `FlawMeta` manipulation but if we would use it in the future then this validation would not be 100% safe.

Reply:
I took it into consideration, but realistically you cannot create a `FlawMeta` without assigning it a flaw since the field is required, so you would have to create a `FlawMeta` with some `Flaw` already assigned to it, and then pass it to some other flaw through `Flaw.meta.set()`... I can implement it but not sure if it even makes sense

Reply:
Yeah, I am aware of that, I was just seeing it as a possible corner case when technically someone is able to manually force this into data and omit the validation. But probably no big deal, it is a really specific case.

Reply:
So I did some more research last night, and well the problem is that `set` and other `RelatedManager` methods do a direct SQL query to perform their actions (to be efficient) and that means that they skip save (but it's still saved into the db regardless).

I thought that maybe we can somehow override these `RelatedManager` methods but it seems like it's not possible [1] since the `RelatedManager` is created dynamically.

The only "solution" is using the `bulk=False` parameter when calling any of these methods [2], but it's not something we can enforce (technically speaking we can but that's more complicated / another story) so it'd have to be by convention, and at that point we might as well convene to simply not use `set()` and `add()` for `Flaw.meta`.

[1] https://stackoverflow.com/questions/4694767/overriding-djangos-relatedmanager-methods#comment5221073_4695692
[2] https://docs.djangoproject.com/en/4.0/ref/models/relations/#django.db.models.fields.related.RelatedManager.add

Reply:
That said, I updated the tests to use `add` and `set` with `bulk=False` to confirm that the validation works in that case too

Reply:
Wow, I haven't thought that this would be that complicated. So I would say that the best idea is probably to just not use the `set()` and `add()` for `Flaw.meta` and this would prevent missing the validation.

Reply:
Good to think about it in the future. Just not to use those methods.